### ISSUE TITLE
Use .debug_aranges

### ISSUE THREAD

**Issue by JunyiXie** [gimli-rs/addr2line#95]
give a sample, thanks.

**Comment by philipc** [comment]
Not sure what you mean by "use it alone". There's sample usage in the [example](https://github.com/gimli-rs/addr2line/blob/master/examples/addr2line.rs), the [tests](https://github.com/gimli-rs/addr2line/tree/master/tests) and also in the [backtrace-rs crate](https://github.com/alexcrichton/backtrace-rs/blob/master/src/symbolize/gimli.rs).

**Comment by JunyiXie** [comment]
Like using addr2line. In terminal.

**Comment by philipc** [comment]
You'll need to install rust, checkout a copy of the addr2line source code, and build the example with `cargo build --examples --release`. The binary will be at `target/release/examples/addr2line`.

**Comment by JunyiXie** [comment]
Does this tool support dsym?

**Comment by philipc** [comment]
Yes, but you need to specify the dsym on the command line instead of the executable.

**Comment by JunyiXie** [comment]
thanks.
How to do it? 
use --help, don't find the relevant information.

**Comment by philipc** [comment]
I'm not able to run it right now to test, but something like `echo 0x1234 | addr2line -e <path-to-dsym>`.

**Comment by JunyiXie** [comment]
thanks. It's work well.

**Comment by JunyiXie** [comment]
```
21:05:03 › echo 0x0000b1e7 | ./addr2line  -s -i -f -p -a -e /Users/xiejunyi/Desktop/atosl/atosl/test/res/CrashTest3Dwarf.thin
0x000000000000b1e7: -[QDCViewController sendCrash:] ??:0
```
Is there a problem with the line number and file name?

**Comment by JunyiXie** [comment]
yes. It's from there.  x86_64 file：
```
echo 0x00000001000010a0 | ./addr2line  -s -i -f -p -a -e /Users/xiejunyi/Desktop/atosl/atosl/test/res/TestMac1.thin
0x00000001000010a0: -[QDCAppDelegate applicationDidFinishLaunching:] ??:0
```

**Comment by philipc** [comment]
Those files have neither `DW_AT_high_pc` nor `DW_AT_ranges` attributes on their `DW_TAG_compile_unit` entries as required by the DWARF spec, so we're ignoring them. They do have a valid `.debug_aranges` though, so maybe the best fix would be to start using that.

**Comment by philipc** [comment]
Fixed in #200
